x-pack/plugins/apm/server/index.ts

```diff
@@ -57,6 +57,8 @@ const configSchema = schema.object({
     defaultValue: 'https://apm-agent-versions.elastic.co/versions.json',
   }),
   enabled: schema.boolean({ defaultValue: true }),
+  serverlessOnboarding: schema.boolean({ defaultValue: false }),
+  managedServiceUrl: schema.string({ defaultValue: '' }),
```

Comment:
__question:__ just for my understanding, is this property just for serverless offering? If so, is there any reason we don't gate it with [`schema.contextRef('serverless')`](https://github.com/elastic/kibana/pull/156837)?

If not only for serverless, is there any public documentation for it? Since we expose this directly to the browser client without requiring authentication, I just want to make sure there is no reason for the admin to put something like `https://secret_username:secret_password@<project-id>.apm.<suffix>` as the value for this property (I doubt, but it's better to double check)?

Reply:
> just for my understanding, is this property just for serverless offering? If so, is there any reason we don't gate it with https://github.com/elastic/kibana/pull/156837?

Yes, this is only for serverless. Is there any harm in not gating it? It will just be undefined in the other environments. 

> Since we expose this directly to the browser client without requiring authentication, I just want to make sure there is no reason for the admin to put something like https://secret_username:secret_password@<project-id>.apm.<suffix> as the value for this property (I doubt, but it's better to double check)?

It's a publicly accessible url matching the kibana instance url. API keys are used for authentication which is handled separately. It is injected by elastic so users cannot accidentally leak private information this way.

Reply:
>  Is there any harm in not gating it? It will just be undefined in the other environments.

There might not be any harm in the short term, but @lukeelmers and @elastic/kibana-core has outlined a number of quality and testing concerns in https://github.com/elastic/kibana/issues/152395. I'll leave it to them to address your specific concerns, but I believe it's important for us to strive for a consistent approach overall.

>  It is injected by elastic so users cannot accidentally leak private information this way.

Thanks, so if it's just for serverless and there is no way for non-Elastic admins to do silly things then it's fine by me :slightly_smiling_face:

Reply:
> > Is there any harm in not gating it? It will just be undefined in the other environments.
> 
> There might not be any harm in the short term, but @lukeelmers and @elastic/kibana-core has outlined a number of quality and testing concerns in #152395. I'll leave it to them to address your specific concerns, but I believe it's important for us to strive for a consistent approach overall.

There are a few benefits to wrapping serverless-only configs in a `schema.contextRef('serverless')` check:
1. It prevents leaking this setting to self-managed users, which ultimately gives us more flexibility to make changes. If we don't include this check, the config effectively becomes a public API, which means we can't remove or break it. If we restrict to serverless-only, we can break this at any time knowing that there are no external users of the setting.
2. It makes explicit the fact that this config is only needed for serverless. This will make maintenance easier for us in the long-run, because the `contextRef` checks are the only global mechanism we have to understand which configs only exist for serverless. If a config doesn't have this gate, we can only assume that it is also needed outside of serverless. It also makes testing easier to be clear about the intended use of a config.
3. In this particular case, it also addresses the concerns @azasypkin raised around sensitive data being inadvertently exposed to the browser. If we gate this config to only serverless, we don't need to worry about users discovering the config and shooting themselves in the foot, because the only folks who will be able to set this config are Elastic admins in the serverless product.

Reply:
@lukeelmers Thank you for the detailed explanation. I will gate the configs with contextRef checks 👍🏼